### PR TITLE
DALA-3170 Deterministic Fake Fixture Generation

### PR DESCRIPTION
**Pull request by MarcTM01**
# Pull Request Deterministic Fake Fixture Generation
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one test exists testing the new feature
  - [x] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [x] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [ ] IF there are changes done to the framework data models or to a database entity class, the following steps were completed in order
  - [ ] A fresh clone of dataland.com is generated (see Wiki page on "OTC" for details)
  - [ ] The feature branch is deployed to clone with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on the clone server
  - [ ] The feature branch is deployed to dev1 with `Reset non-user related Docker Volumes & Re-populate` turned on, and it's verified that the CD run is green  
- [x] ELSE, the new version is deployed to the dev server "dev1" using this branch
  - [x] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on dev server
- [x] Make sure that pull requests in the development tools repository, which are connected to this pull requests, 
are merged

### CODE REVIEW

File: startDevelopmentStack.sh [hunk 1]
```diff
@@ -16,7 +16,15 @@ source ./*github_env.log
 source ./environments/.env.uncritical
 set +o allexport
 
-find ./build-utils/ -name "rebuild*.sh" ! -name "*prod*" ! -name "*test*" ! -name "*backend*" -exec bash -c 'eval "$1" && echo "SUCCESS - execution of $1 was successful" || echo "ERROR - could not execute $1"' shell {} \;
+if [ $# -ge 1 ] && [ "$1" == "parallel" ]; then
+  echo "Building docker images in parallel for unrivaled speed! (Does not work in Windows Git Bash)"
```
[the remark this answers is not in the document]
Comment (MarcTM01) [reply startDevelopmentStack.sh:20]: Not easily. It uses GNU Parallel to run the docker build scripts in parallel (which is a standard linux utility). This utility is neither included in the git bash nor in the powershell. Nevertheless, it may potentially be possible to install GNU Parallel in git bash (But unfortunately not easily). But it does speed things up for me :)

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/Lksg.kt [hunk 2]
```diff
@@ -113,7 +113,7 @@ class Lksg {
 
     private fun sortComplaintRisksInDataset(dataset: LksgData): LksgData {
         val complaintRisksIdentifiedRisks = dataset.governance?.grievanceMechanismOwnOperations
-            ?.complaintsRiskPosition!!.toMutableList()
+            ?.complaintsRiskPosition?.toMutableList() ?: mutableListOf()
```
Comment (marcelrasbachdfine) [inline dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/Lksg.kt:116]: why this change? i though this would be the same in terms of functionality

Comment (MarcTM01) [reply dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/Lksg.kt:116]: No. The previous one crashes when complaintsRiskPosition is null because of the non-null-assertion

File: dataland-frontend/tests/e2e/fixtures/ReproducibilityConfiguration.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+/**
+ * Combines multiple hashcodes similar to how Objects.hash works in java
+ * @param codes the hash codes to combine
+ * @returns the combined hash code
+ */
+function combineHashCodes(...codes: number[]): number {
+  let hash = 1;
+  for (const code of codes) {
+    hash = 31 * hash + code;
```
Comment (marcelrasbachdfine) [inline dataland-frontend/tests/e2e/fixtures/ReproducibilityConfiguration.ts:22]: no magic number error in sonar? :D
